### PR TITLE
Replace uses of `set_sample_data`

### PR DESCRIPTION
**Pull request by unflxw**
[skip changeset]

Do not merge before appsignal/appsignal-elixir#713 and appsignal/appsignal-elixir-plug#20.

### Replace uses of set_sample_data

Since `set_sample_data/3` is deprecated, usages of it have been replaced with `set_params/2` and `set_environment/2`.

### Point to the integration's branch

TODO:
- [ ] Rebase out this commit before merging, after the `deprecate-set-sample-data` and `replace-set-sample-data` branches have been merged to `main`.

### CODE REVIEW

File: mix.exs
Comment (tombruijn) [inline mix.exs:61]: I think the future it's okay not to commit this and have the build fail until the PR this is blocked on is merged. Reduces the amount of rebasing and force pushing.

**Review by tombruijn — APPROVED** [review]

File: mix.exs
Comment (unflxw) [reply mix.exs:61]: Noted! I mostly do it for my own peace of mind, though. Seeing the green tick in the CI check soothes me.

File: mix.exs
Comment (tombruijn) [inline mix.exs:60]:
```suggestion
        ">= 2.2.8 and < 3.0.0",
```

Probably need to change this. See https://github.com/appsignal/appsignal-elixir/pull/713#issuecomment-965345406

**Review by luismiramirez — APPROVED** [review]

**Review by jeffkreeftmeijer — CHANGES_REQUESTED** [review]
Looks good, but this needs to target the yet-to-be-released :appsignal_plug version, which depends on the version of :appsignal with these functions added.

File: mix.exs
Comment (unflxw) [reply mix.exs]: Done! This is `appsignal-phoenix`, though, which depends on `appsignal-plug`, so I bumped `appsignal-plug`'s version number as well so we could target it here: it's `2.0.9`, not `2.2.8`.

**Review by jeffkreeftmeijer — CHANGES_REQUESTED** [review]
We’ll need to remove the `github` key from the :appsignal_plug dependency. Aside from that, this one’s good to go.

File: lib/appsignal_phoenix/live_view.ex
Comment (tombruijn) [inline lib/appsignal_phoenix/live_view.ex:23]: I think this looks fine. We can't chain it anymore on the span, like we did previously below, but is that necessary?

But maybe @jeffkreeftmeijer has a stronger opinion about this?

File: lib/appsignal_phoenix/live_view.ex
Comment (tombruijn) [inline lib/appsignal_phoenix/live_view.ex:27]: I see there's no `set_error` function on the Elixir tracer either. Like we did for the Node.js integration. We should add that in another PR. Let's create an issue for that.

**Review by tombruijn — APPROVED** [review]

**Comment by unflxw** [comment]
Closing. See [appsignal/appsignal-elixir#713](https://github.com/appsignal/appsignal-elixir/pull/713#issuecomment-979877206).

**Pull request closed without merging**
